src/js/report_form.js

```diff
@@ -41,6 +52,60 @@
     }
   }
 
+  /**
+   * Converts the specified time to the timezone provided
+   *
+   * @param {string} time string representing HH:mm string
+   * @param {string} tz which timezone to convert to
+   */
+  function convertToTimezone(time, tz) {
+    if (time === "") {
+      return "";
+    }
+    let dateTimeObj = moment.utc(`${defaultDate} ${time}`);
+    dateTimeObj.tz(tz);
+    return dateTimeObj;
+  }
+  /**
+   * Gets current timezone from local storage
+   * @returns {datetime} either the browser
+   *   timezone or manually selected timezone
+   */
+  function getSelectedTimezone() {
+    const manualTz = localStorage.getItem("chosen-timezone");
+    const selectedTz = localStorage.getItem("selected-timezone");
+    return selectedTz || manualTz;
+  }
+  /**
+   * Gets current timezone from local storage
+   * @param {scheduleType} type of schedule so we know whether
+   *   to update the weekdayinput
+   */
+  function convertTimesToLocalTimezone(scheduleType) {
+    scheduleTimezoneInput.value = getSelectedTimezone();
+    let convertedTime = convertToTimezone(
+      scheduleTimeInput.value,
+      scheduleTimezoneInput.value
+    );
+
+    scheduleTimeInput.value =
+      convertedTime !== "" ? convertedTime.format("HH:mm") : "";
+    scheduleTimeInput.dispatchEvent(new Event("change"));
+
+    if (scheduleType === "weekly" && scheduleWeekDayInput.value !== "") {
+      const offset = convertedTime.day() - moment(defaultDate).day();
+
+      if (offset != 0) {
+        const currDayOfWeek =
+          offset > 0
+            ? scheduleWeekDayInput.value + 1
+            : scheduleWeekDayInput.value - 1;
```

Comment:
couple more things here.
* `scheduleWeekDayInput.value` is type of `string` but you're doing arithmetical operations with it. It works with `-` because it's automatically [coerced](https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion) to `number` but not with `+`
`"3" + 1` = `"31"`
* something wrong with logic here.
  I can't set schedule to the evening time when I'm in "New York" timezone for example.
  Setting it to be Friday 8pm and when I get back to it it shows Thursday 8pm even though i haven't touched timezone setting

Reply:
this is a problem actually on the backend. i have a fix for this in a sec